src/restrictingGuards.ts

```diff
@@ -0,0 +1,97 @@
+import {checkerToGuard} from './Checker';
+import {thenGuard, andGuard, orGuard} from './combinators';
+import {ReasonGuard} from './ReasonGuard';
+import {isNumber, isSymbol, isString} from './primitiveGuards';
+
+export const numberIsInteger = checkerToGuard<number, number>((input: number) => {
+	if (!Number.isInteger(input)) {
+		throw new Error(`${input} is not an integer`);
+	}
+	return `${input} is an integer`;
+});
+
+export const numberIsLessThan = (maximum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input >= maximum) {
+			throw new Error(`${input} >= ${maximum}`);
+		}
+		return `${input} < ${maximum}`;
+	});
+
+export const numberIsGreaterThan = (minimum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input <= minimum) {
+			throw new Error(`${input} <= ${minimum}`);
+		}
+		return `${input} > ${minimum}`;
+	});
+
+export const numberIsAtMost = (maximum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input > maximum) {
+			throw new Error(`${input} > ${maximum}`);
+		}
+		return `${input} <= ${maximum}`;
+	});
+
+export const numberIsAtLeast = (minimum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input < minimum) {
+			throw new Error(`${input} < ${minimum}`);
+		}
+		return `${input} >= ${minimum}`;
+	});
+
+export const openIntegerRange =
```

Comment:
This name doesn't match the others ... or its impl.  This is implemented as `closedIntegerRange`, and would be better named IMO `numberBetweenInclusive` (or `numberBetweenExclusive` for the actual open range version).

Arguably there ought to be a fully general version that takes both endpoints and a bool for each whether it should be open or closed.

Reply:
Yeah, I went through this same thing. The implementation *is* correct -- at-least and at-most are open (GTE and LTE). I was trying to come up with a clever syntax but couldn't find the right one.